src/cosim/slave_simulator.cpp

```diff
@@ -79,6 +79,38 @@ struct get_variable_cache
             }
         }
     }
+
+    serialization::node export_state() const
+    {
+        assert(references.size() == originalValues.size());
+        assert(std::all_of(modifiers.begin(), modifiers.end(), [](auto m) { return !m; }));
```

Comment:
Other option is to print warning if manipulators are currently applied and then continue exporting?

Reply:
Yeah, this could be a good option too. The main problem is that I am very unsure of what will happen if manipulators are active – under which circumstances it will work just fine, when it will subtly fail, and when it will fail spectacularly. I suspect that it will produce wrong results in the majority of cases, so I thought it was safer to make it an assertion rather than a warning. I think the whole “modifier” concept should be completely redesigned with a proper interface (à la `simulator`, `observer`, etc.) for this to work. It is just too easy to hide statefulness and aliasing behind a `std::function` object.